rasa/nlu/extractors/duckling_entity_extractor.py

```diff
@@ -172,41 +171,31 @@ def _reference_time_from_message(message: Message) -> int:
                     "duckling. Error: {}".format(message.time, e)
                 )
         # fallbacks to current time, multiplied by 1000 because duckling
-        # requires the reftime in miliseconds
+        # requires the reftime in milliseconds
         return int(time.time()) * 1000
 
-    def process(self, message: Message, **kwargs: Any) -> None:
+    def process(self, messages: List[Message]) -> List[Message]:
+        if self._url() is None:
+            rasa.shared.utils.io.raise_warning(
+                "Duckling HTTP component in pipeline, but no "
+                "`url` configuration in the config "
+                "file nor is `RASA_DUCKLING_HTTP_URL` "
+                "set as an environment variable. No entities will be extracted!",
+                docs=DOCS_URL_COMPONENTS + "#DucklingEntityExtractor",
+            )
+            return messages
 
-        if self._url() is not None:
+        for message in messages:
             reference_time = self._reference_time_from_message(message)
             matches = self._duckling_parse(message.get(TEXT), reference_time)
             all_extracted = convert_duckling_format_to_rasa(matches)
```

Comment:
Is it worth recording that we'll need to migrate these functions (this and `filter_irrelevant_entities`) in the deprecation ticket for the non-graph version of this?

Reply:
oh good catch. That's actually an error 🤦🏻

Reply:
So `convert_duckling_format_to_rasa` is fine since it's a function but the `filter_irrevalent_entities` one was wrong